Header component

DavidBronfen opened the pull request:
#7

DavidBronfen commented:
Trying to see the `<p> header works!</p>` in the app, the HeaderComponent has been included in the `app.module` file.

nirgn975 commented on client/src/app/app.module.ts:
```diff
@@ -7,10 +7,12 @@ import { HttpModule } from '@angular/http';
   declarations: [
-    AppComponent
+    AppComponent,
+    HeaderComponent
```
Add a trailing comma.

nirgn975 commented:
> Trying to see the <p> header works!</p> in the app, the HeaderComponent has been included in the app.module file.

@DavidBronfen You should also use this component in some other component. like write 
`<app-header></app-header>` in `app.component.html` file.

DavidBronfen commented:
Anonymous user icon looks pixelated 
The icon:
![anonymous_chief](https://cloud.githubusercontent.com/assets/12427216/23737332/0b812b92-049b-11e7-929a-99cb57d9a692.png)
A screenshot of the header:
![image](https://cloud.githubusercontent.com/assets/12427216/23737347/2c2fd9ba-049b-11e7-97d3-2f535fbb64ba.png)

DavidBronfen commented:
Removing the default test form the `app.po.ts` and `app.e2e-spec.ts`.

DavidBronfen commented:
@nirgn975 any suggestion?

nirgn975 commented:
@DavidBronfen Done.
Please look at [the code I changed](https://github.com/DavidBronfen/Retrogusto/pull/9/commits/9a8c4cfce6cf74beb3db87562b6abf1af82c39be) and try to understand why this was the problem.

DavidBronfen commented:
@nirgn975 I think I understand all the changes you've been made, 
Deleted the tests and added the `NO_ERRORS_SCHEMA` (that i don't realy know what is it but i guess it's dealing with the errors of the test)

DavidBronfen commented:
@nirgn975 Do you think it would be better to add all the data files (images, icons, etc.) in his own Pull request instead of loading it here?

nirgn975 commented:
When testing a component that has sub-components you have a few options:

1. Import the module that sets up the component and sub-components. Problem is you instantiate the real sub-components and their dependencies.
2. Mock sub-components. This can be tedious depending on how many sub-components it has.
3. Use `NO_ERRORS_SCHEMA` and ignore sub-components.

Given that most of the time you only care about the current component, we use here `NO_ERRORS_SCHEMA`. We care only about the current component because we test the other components separately.

DavidBronfen commented:
Waiting for #15 to be merged

nirgn975 commented:
@DavidBronfen Is this ready to review?

DavidBronfen commented:
@nirgn975 Will be tonight.

DavidBronfen commented:
@nirgn975 This PR should be ready for review.
I've just changed the logo image to `svg`.

nirgn975 commented:
@DavidBronfen But the tests are still failing \:
Actually I'll have more time today, so I will fix those and let's merge this PR already - it's been far too long in the works..
